src/components/views/rooms/LinkPreviewWidget.js

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2016 OpenMarket Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+'use strict';
+
+var React = require('react');
+
+var sdk = require('../../../index');
+var MatrixClientPeg = require('../../../MatrixClientPeg');
+var ImageUtils = require('../../../ImageUtils');
+var Modal = require('../../../Modal');
+
+var linkify = require('linkifyjs');
+var linkifyElement = require('linkifyjs/element');
+var linkifyMatrix = require('../../../linkify-matrix');
+linkifyMatrix(linkify);
+
+module.exports = React.createClass({
+    displayName: 'LinkPreviewWidget',
+
+    propTypes: {
+        link: React.PropTypes.string.isRequired,
+        mxEvent: React.PropTypes.object.isRequired,
+        onCancelClick: React.PropTypes.func,
+        onWidgetLoad: React.PropTypes.func,
+    },
+
+    getInitialState: function() {
+        return {
+            preview: null
+        };
+    },
+
+    componentWillMount: function() {
+        MatrixClientPeg.get().getUrlPreview(this.props.link, this.props.mxEvent.getTs()).then((res)=>{
+            this.setState({ preview: res });
+            this.props.onWidgetLoad();
```

Comment:
you probably want to pass this as the callback for `setState`, so that it is called once the DOM is updated.